conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandlerTest.java

```diff
@@ -80,6 +86,74 @@ public void handlesServiceException() throws IOException {
                 .contains("\"parameters\":{\"foo\":\"bar\"}}");
     }
 
+    @Test
+    public void checkServiceExceptionArgumentsSerializedWithToString() throws IOException {
+        record Argument(
+                String field1, Optional<Integer> maybeField2, List<String> collectionField, String nullString) {}
+        exception = new ServiceException(
+                ErrorType.CONFLICT,
+                SafeArg.of("arg", new Argument("foo", Optional.of(42), List.of("bar", "baz"), null)));
```

Comment:
thought on extracting `Argument` to a variable, that way instead of encoding `String expectedSerializedArg = <string literal>` we can assert against `argument.toString()`? that keeps the code a bit easier to follow, and ensures that if the toString changes slightly, the test won't fail.

Reply:
That was cleaner!